File: ios/OffshoreFx/OffshoreFxModel.swift
//
//  OffshoreFxModel.swift
//  Runner
//
//  Created by Saulo Nascimento on 17/01/26.
//

import Foundation

public struct OffshoreFxModel {
    public let accountNumber: String

    
    public init(accountNumber: String) {
        self.accountNumber = accountNumber
    }

    public static func fromMap(_ map: [String: Any]?) -> OffshoreFxModel? {
        guard let map = map else { return nil }
        
        return OffshoreFxModel(
            accountNumber: map["accountNumber"] as? String ?? ""
        )
    }
}

File: android/offshorefx/src/main/java/com/saulo/dev/offshorefx/OffshoreFxModel.kt
package com.saulo.dev.offshorefx

data class OffshoreFxModel(val accountNumber: String) {
    companion object {
        fun fromMap(map: Map<String, Any>?): OffshoreFxModel? {
            if (map == null) return null
            return OffshoreFxModel(
                accountNumber = map["accountNumber"] as? String ?: "",
            )
        }
    }
}
